lib/msf/core/auxiliary/http_crawler.rb

```diff
@@ -243,7 +243,7 @@ def crawl_target(t)
   # Specific module implementations should redefine this method
   # with whatever is meaningful to them.
   def crawler_process_page(t, page, cnt)
-    msg = "[#{"%.5d" % cnt}/#{"%.5d" % max_page_count}]    #{page.code || "ERR"} - #{@current_site.vhost} - #{page.url}"
+    msg = "[#{"%.5d" % cnt}/#{"%.5d" % max_page_count}]    #{page ? page.code || "ERR" : "ERR"} - #{@current_site.vhost} - #{page.url}"
     case page.code
```

Comment:
Just confirming, if page is `nil` - would this break as well?

Reply:
Yeah looks like this would break. Still need to find a solution for this case as well.

Reply:
After further inspection, what we really should be doing is failing entirely if `page` is `nil`. The code relies on `page` in multiple locations to the extent that if its `nil` we really shouldn't even be trying to continue; we have a critical expectation failure and we should be bailing at that point.

Reply:
Think I fixed this case in particular with https://github.com/rapid7/metasploit-framework/pull/16668/commits/d20fa45f7a1375956fef374d50f190f2b4dfbf1e